pkg/grub/rootfs-installer.cfg

```diff
@@ -0,0 +1,491 @@
+}
+
+function set_arm64_baremetal {
+   set_generic
+   set_arm64
+   set_global dom0_platform_tweaks " "
+
+   # experimental feature where we're trying to find known device trees on ARM
+   # note that we would rather rely on UEFI providing device tree information,
+   # but sometimes this can be useful and besides it is only triggered on well
+   # known IDs of products below:
+   smbios -t 1 -s 0 --set smb_vendor
+   if [ "$smb_vendor" = "nvidia" ]; then
+      smbios -t 1 -s 5 --set smb_product
+      if [ "$smb_product" = "p3450-0000" ]; then
+         set_to_existing_file devicetree /boot/dtb/nvidia/tegra210-p3450-0000.dtb
+      fi
+   fi
+   if [ "$smb_vendor" = "raspberrypi" ]; then
+      smbios -t 1 -s 5 --set smb_product
+      if [ "$smb_product" = "rpi" ]; then
+         set_to_existing_file devicetree /boot/dtb/broadcom/bcm2711-rpi-4-b.dtb
+      elif [ "$smb_product" = "uno-220" ]; then
+         set_to_existing_file devicetree /boot/dtb/broadcom/raspberrypi-uno-220.dtb
+      fi
+   fi
+   if [ "$smb_vendor" = "Huawei" ]; then
+      smbios -t 1 -s 5 --set smb_product
+      # turns out smb_product contains non-printable characters
+      # this makes grepping for XR320 our best option
+      if regexp -- "XR320" "$smb_product"; then
+         set_global dom0_console "console=ttyAMA0,115200n8"
+         set_global dom0_platform_tweaks "pcie_aspm=off pci=pcie_bus_perf"
+      fi
+   fi
+   if [ "$smb_vendor" = "rockchip" ]; then
+      smbios -t 1 -s 5 --set smb_product
+      if [ "$smb_product" = "evb_rk3399" ]; then
+         set_to_existing_file devicetree /boot/dtb/rockchip/rockchip-evb_rk3399.dtb
+         set_global dom0_console "console=ttyS2 console=ttyS2,115200n8"
+         set_global dom0_platform_tweaks "video=HDMI-A-1 coherent_pool=1M drm.debug=0x04"
+      fi
+   fi
+   if [ "$smb_vendor" = "fsl" ]; then
+      smbios -t 1 -s 5 --set smb_product
+      if [ "$smb_product" = "imx8mq-evk" ]; then
+         set_to_existing_file devicetree /boot/dtb/freescale/imx8mq-evk.dtb
+         set_global dom0_console "console=ttymxc0,115200 earlycon=ec_imx6q,0x30860000,115200"
+      fi
+   fi
+   if [ "$smb_vendor" = "phytec" ]; then
+      smbios -t 1 -s 5 --set smb_product
+      if [ "$smb_product" = "pollux" ]; then
+         set_to_existing_file devicetree /boot/dtb/freescale/imx8mp-phyboard-pollux-rdk.dtb
+         set_global dom0_console "console=tty0 console=hvc0 console=ttymxc0,115200 earlycon=ec_imx6q,0x30860000,115200"
+      fi
+   fi
+   if [ "$smb_vendor" = "advantech" ]; then
+      smbios -t 1 -s 5 --set smb_product
+      if [ "$smb_product" = "imx8mp_rsb3720a1" ]; then
+         set_to_existing_file devicetree /boot/dtb/freescale/imx8mp-rsb3720-a1.dtb
+         set_global dom0_console "console=tty0 console=hvc0 console=ttymxc2,115200 earlycon=ec_imx6q,0x30880000,115200"
+      fi
+   fi
+   # Nvidia's Jetson Xavier NX developer kit (variant identified by the Asset-Tag)
```

Comment:
All the custom configs here are already defined in the regular rootfs.cfg file, you can move all of them to a common config file and just use it on both rootfs.cfg and rootfs-installer.cfg, so we don't have duplicated definitions which is really hard to maintain.....

Reply:
I think he addressed this.